### ISSUE TITLE
Problem when instantiation a controller object

### ISSUE THREAD

**Issue by maiconkist** [wishful-project/framework#4]
The following error ocurred when executing ANY of the examples provided in the examples/ folder:

```
  File "./wishful_simple_controller", line 46, in <module>
    controller = wishful_controller.Controller(dl="tcp://127.0.0.1:8990", ul="tcp://127.0.0.1:8989")
  File "/home/kistm/.local/lib/python3.5/site-packages/wishful_controller/controller.py", line 122, in __init__
    self.radio = builder.create_radio()
  File "/home/kistm/.local/lib/python3.5/site-packages/wishful_framework/controller/upis_builder.py", line 146, in create_radio
    function = self.perpare_function(method, function)
  File "/home/kistm/.local/lib/python3.5/site-packages/wishful_framework/controller/upis_builder.py", line 122, in perpare_function
    function = compile(code, 'My_Code', 'exec')
  File "My_Code", line 1
SyntaxError: duplicate argument 'self' in function definition
```

I've solved it with the following changes (basically I check if there is  a 'self' in the string before trying to add the 'self'):

```
diff --git a/wishful_framework/controller/upis_builder.py b/wishful_framework/controller/upis_builder.py
index 2913a53..95eb877 100644
--- a/wishful_framework/controller/upis_builder.py
+++ b/wishful_framework/controller/upis_builder.py
@@ -109,7 +109,11 @@ class UpiBuilder(object):
     def perpare_function(self, name, function):
         sig = get_method_sig(function)
         sig = sig.split("(", 1)
-        sig = sig[0] + "(self, " + sig[1]
+
+        if not 'self' in sig[1]:
+       	    sig = sig[0] + "(self, " + sig[1]
+        else:
+       	    sig = sig[0] + "(" + sig[1]
         code = "def {}: pass".format(sig)
         myGlobals = {}
         #exec(code, myGlobals
```

**Comment by pruckebusch** [comment]
Hi Maicon,

indeed the problem you are describing is correct.
It was introduced while adding the meta-models to the UPIs. As they are defined as non-abstract classes they are also callable. For this reason they are also considered as UPIs.
The solution you propose will probably work but I would prefer a more cleaner solution. I tested the following line 
if inspect.isfunction(getattr(module, method)):
which should do the trick.

I will create a branch and do a pull request.

Kind regards,

Peter

**Comment by mchwalisz** [comment]
Is it already fixed with #5 ?

**Comment by pruckebusch** [comment]
Indeed, I will close the issue.
